### ISSUE TITLE
False negative on the `Style/Next` cop

### ISSUE THREAD

**Issue by ydakuka** [rubocop/rubocop#12448]
## Actual behavior

I have [the following code](https://docs.rubocop.org/rubocop/cops_style.html#stylenext):

```ruby
def my_method
  [1, 2].each do |a|
    if a == 1
      puts a
    end
  end
end
```

I run rubocop:
```console
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop app/models/test.rb
Inspecting 1 file
.

1 file inspected, no offenses detected
```

## RuboCop version

```console
ydakuka@yauhenid:~/Work/project$ bin/rails_docker rubocop -V
1.58.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 2.7.8) [x86_64-linux]
  - rubocop-capybara 2.19.0
  - rubocop-factory_bot 2.24.0
  - rubocop-performance 1.19.1
  - rubocop-rails 2.22.2
  - rubocop-rake 0.6.0
  - rubocop-rspec 2.25.0
  - rubocop-thread_safety 0.5.1
```
